src/Rule/AbstractCharCase.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ *
+ * This file is part of Aura for PHP.
+ *
+ * @license http://opensource.org/licenses/bsd-license.php BSD
+ *
+ */
+namespace Aura\Filter\Rule;
+
+use Aura\Filter\Exception;
+
+/**
+ *
+ * Abstract rule for character case filters; supports the `mbstring`
+ * extension.
+ *
+ * @package Aura.Filter
+ *
+ */
+abstract class AbstractCharCase extends AbstractStrlen
```

Comment:
I think you're asking where the `substr` method resides?

`AbstractStrlen` is preexisting, and contains this method. It's used for the previously existing 'string length' functions. 

The 2nd commit, which adds `ufcirst` and `lcfirst` functions, changes the `AbstractCharCase` class to extend `AbstractStrlen`, since there's not `mbstring` functions which can do uc/lcfirst.

Perhaps there's a more clear solution? Replicate the code? Traits? Different naming convention?